### ISSUE TITLE
General hook for evaluation on train/dev/test

### ISSUE THREAD

**Issue by tdmeeste** [uclmr/jack#95]
Extended `quebap.sisyphos.TestAllHook` to `quebap.sisyphos.EvalHook`, to be called during training or afterwards, on train/dev/test data.  Implemented various metrics (see below), most still need to be tested!
So far, tested on two cases:
(1) SNLI data: fixed number of classes, single correct answer (tested with indices and binary vectors as targets/predictions)
(2) TacKBP data (not yet with the quebap input file):  fixed number of classes, 0 or more correct ones (requires binary vectors as predictions and targets).

**Note: in time, `EvalHook` should replace `AccuracyHook`, because it has more general support ( e.g. in case (2) AccuracyHook fails), and it covers more metrics besides accuracy.**

I added some documentation: 

```python
       """
        Initialize EvalHook object.
        Calling the hook prints calculated metrics to stdout, and returns targets, predictions, and a metrics dict.

        Args:
            batches:
                iterator / generator of batches; assumed each batch is a proper feed_dict in case placeholders=None
                otherwise paired with the placeholders to form feed_dicts.
            logits:
                tf op with logits
            predict:
                tf op that returns binary predictions, either for each instance as the index of the predicted answer (if unique)
                otherwise as a tensor of shape (batch_size, num_candidates), in which each prediction is a
                num_candidates-long binary vector (with 0 or more 1's and the rest 0).
                (num_candidates may vary over batches; in that case the metrics macro- microP/R/F1 are not available,
                but Acc and MRR are)
            target:
                binary targets; either for each instance as index of correct answer (if unique),
                or as binary vectors (with 0 or more 1's for correct answers), similar to predict.
            at_every_epoch:
                evaluated each at_every_epoch epochs. Note that calling EvalHook requires loss-argument to be 0,
                as automatically done post-epoch in quebap.sisyphos.train, hence always called after finishing the epoch.
            placeholders:
                placeholders, in case batches does not generate feed_dicts.
            metrics: list with metrics;
                default: [], which will return all available metrics (dep. on type of problem)
                Options are currently:
                'Acc', 'MRR', 'microP', 'microR', 'microF1', 'macroP', 'macroR', 'macroF1', 'precision', 'recall', 'F1'.
                Note: mostly not all the metrics are available, but deciding on sensible metrics is left to the user;
                    E.g.: if the number of candidates remains constant, micro- and macro-metrics will be returned,
                    even if the 'labels' for the actual candidates vary over instances.
            summary_writer:
                optional summary_writer
            print_details:
                if True, prints for each instance:
                target, prediction, logits, and whether the prediction is entirely correct
            print_to:
                if a filename is specified, appends the details per instance (as for print_details) to that file.
            msg:
                print this message before the evaluation metrics; useful to distinguish between calls of en EvalHook on train/dev/test data.
        """
```
